src/main/java/calculator/model/StringParser.java

```diff
@@ -0,0 +1,48 @@
+package calculator.model;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class StringParser {
+    public List<Integer> extractNumbers(String str) {
+        Validation inputValidation = new Validation();
+        List<String> stringList;
+        String delimiter = findDelimiter(str);
+
+        str = trimInputString(str);
+
+        stringList = Arrays.stream(str.split(delimiter)).map(s -> s.isEmpty() ? "0" : s).collect(Collectors.toList());
+
+        if (!inputValidation.isValidInput(stringList)) {
+            throw new IllegalArgumentException("입력된 문자열에 유효하지 않은 값이 포함되어 있습니다.");
+        }
+
+        List<Integer> numbers = stringList.stream().map(Integer::parseInt).collect(Collectors.toList());
```

Comment:
Java 16 이상부터는 collect(Collectors.toList()) 대신 아래와 같이 toList() 메서드를 사용할 수 있습니다!

```suggestion
        List<Integer> numbers = stringList.stream().map(Integer::parseInt).toList(); 
```

리스트를 수정해야 한다면 collect(Collectors.toList()) 를 사용할 수 있겠지만,
위의 경우는 toList() 를 써서 immutable 한 리스트를 만드는 것이 입력 데이터의 잘못된 변경을 막을 수 있어 좋아보여요!